### PR TITLE
Turn on -Wconversion in some tests, shell, jsontlv.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**

### CODE REVIEW

**Review by cliffamzn — APPROVED** [review]

File: src/inet/BasicPacketFilters.h [hunk 1]
```diff
@@ -42,7 +42,7 @@ class DropIfTooManyQueuedPacketsFilter : public chip::Inet::EndpointQueueFilter
      *
      * @param maxAllowedQueuedPackets - max number of pending-in-queue not yet processed predicate-matching packets
      */
-    DropIfTooManyQueuedPacketsFilter(size_t maxAllowedQueuedPackets) : mMaxAllowedQueuedPackets(maxAllowedQueuedPackets) {}
+    DropIfTooManyQueuedPacketsFilter(int maxAllowedQueuedPackets) : mMaxAllowedQueuedPackets(maxAllowedQueuedPackets) {}
```
Comment (arkq) [inline src/inet/BasicPacketFilters.h:45]: Why this was downgraded to signed integer? Is there any reason for this argument to be negative? Maybe instead we should change `mNumQueuedPackets` and `mMaxAllowedQueuedPackets` to `atomic_size_t`?

Comment (bzbarsky-apple) [reply src/inet/BasicPacketFilters.h:45]: I tried that initially, but that flows into a ton of code that works with signed integers only, and I was trying to not boil too many oceans. If you feel strongly about this, I can try it again.

**Review by emargolis — APPROVED** [review]

File: src/inet/BasicPacketFilters.h (hunk 1 above)
Comment (bzbarsky-apple) [reply src/inet/BasicPacketFilters.h:45]: @arkq And in particular, we have code like this:
```
        // If we ever go negative, we have mismatch ingress/egress filter via predicate and
        // device may eventually starve.
        VerifyOrDie(numQueuedPackets >= 0);
```
which will stop checking what it's meant to be checking if all these numbers get converted to `size_t`....

Comment (bzbarsky-apple) [reply src/inet/BasicPacketFilters.h]: Converted them to `atomic_size_t`.

**Review by emargolis — APPROVED** [review]

**Review by vivien-apple — APPROVED** [review]

**Pull request merged**